SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/StaticSubobjectsTest/StaticSubobjectsTest.cpp

```diff
@@ -0,0 +1,353 @@
+			{
+				RequireTrue(PlayerPawn == GetFlowPlayerController()->AcknowledgedPawn, TEXT("The client should possess the pawn."));
+				FinishStep();
+			}
+		},
+		StepTimeLimit);
+
+	// Step 3
+	CheckClientNumberComponentsOnTestActorWithoutWait(InitialNumComponents);
+
+	// Step 4
+	ServerSetIntProperty(1);
+
+	// Step 5
+	CheckClientCanNotSeeIntPropertyWithWait(1);
+
+	// Step 6 - Server removes a static component
+	AddStep(TEXT("StaticSubobjectsTestServerDestroyOneSubobject"), FWorkerDefinition::Server(1), nullptr, [this]() {
+		AssertEqual_Int(GetNumComponentsOnTestActor(), InitialNumComponents,
+						TEXT("AStaticSubobjectTestActor should have the initial number of components"));
+
+		DestroyOneNonRootComponent();
+
+		AssertEqual_Int(GetNumComponentsOnTestActor(), InitialNumComponents - 1,
+						TEXT("AStaticSubobjectTestActor should have had one component removed"));
+		FinishStep();
+	});
+
+	// Step 7
+	MoveClientPawn(PawnRemoteLocation);
+
+	WaitForRelevancyUpdateIfInNative();
+
+	// Step 8
+	CheckClientSeeIntProperty(1);
+
+	// Step 9
+	CheckClientNumberComponentsOnTestActorWithoutWait(InitialNumComponents);
+
+	// Step 10
+	MoveClientPawn(PawnSpawnLocation);
+
+	WaitForRelevancyUpdateIfInNative();
+
+	// Step 11
+	ServerSetIntProperty(2);
+
+	// Step 12
+	CheckClientCanNotSeeIntPropertyWithWait(2);
+
+	// Step 12 - Remove another component from the subobject. This time the client should see the removal once the actor is back in its
+	// interest
+	AddStep(TEXT("StaticSubobjectsTestServerDestroySecondSubobject"), FWorkerDefinition::Server(1), nullptr, [this]() {
+		AssertEqual_Int(GetNumComponentsOnTestActor(), InitialNumComponents - 1,
+						TEXT("AStaticSubobjectTestActor should have one less than the initial number of components on the server."));
+		DestroyOneNonRootComponent();
+		AssertEqual_Int(GetNumComponentsOnTestActor(), InitialNumComponents - 2,
+						TEXT("AStaticSubobjectTestActor should have had another component removed"));
+		FinishStep();
+	});
+
+	// Step 14
+	CheckClientNumberComponentsOnTestActorWithWait(InitialNumComponents);
+
+	// Step 15
+	MoveClientPawn(PawnRemoteLocation);
+
+	// Step 16
+	CheckClientSeeIntProperty(2);
+
+	// Step 17
+	CheckClientNumberComponentsOnTestActorWithWait(InitialNumComponents - 1);
+
+	// Step 18 - Server Cleanup.
+	AddStep(TEXT("StaticSubobjectsTestServerCleanup"), FWorkerDefinition::Server(1), nullptr, [this]() {
+		// Possess the original pawn, so that the spawned character can get destroyed correctly
+		ASpatialFunctionalTestFlowController* ClientOneFlowController = GetFlowController(ESpatialFunctionalTestWorkerType::Client, 1);
+		APlayerController* PlayerController = Cast<APlayerController>(ClientOneFlowController->GetOwner());
+
+		if (AssertIsValid(PlayerController, TEXT("PlayerController should be valid")))
+		{
+			PlayerController->Possess(ClientDefaultPawn);
+			FinishStep();
+		}
+	});
+}
+
+void AStaticSubobjectsTest::MoveClientPawn(FVector& ToLocation)
```

Comment:
What are thoughts on functionising steps that are used multiple times like this? On one hand it makes the test easier to read, on the hand it takes a bit longer to figure out what step the test failed on, given the test's logs.

could pass in the step name I guess